File: scripts/analyze_cgu.py
import sys
import PyPDF2
import re

def analyze_pdf(file_path):
    print(f"\n📄 Analyse de: {file_path}\n")
    
    try:
        with open(file_path, 'rb') as file:
            pdf_reader = PyPDF2.PdfReader(file)
            num_pages = len(pdf_reader.pages)
            
            print(f"📊 Total pages: {num_pages}\n")
            
            # Extraire tout le texte
            full_text = ""
            for page_num in range(num_pages):
                page = pdf_reader.pages[page_num]
                full_text += page.extract_text()
            
            print(f"📝 Caractères extraits: {len(full_text)}\n")
            
            # Rechercher des mots-clés
            keywords = {
                'souscription': r'souscri(ption|re|t)',
                'activation': r'activ(er|ation|é)',
                'conditions': r'condition',
                'accès': r'accès|acces',
                'OAuth': r'oauth',
                'authentification': r'authentification',
                'API': r'\bAPI\b',
                'délai': r'délai|delai',
                'validation': r'validat(ion|e)',
                'acceptation': r'acceptat(ion|e)',
                'CGU': r'\bCGU\b',
                'droit': r'droit',
                'quota': r'quota',
                'limite': r'limite'
            }
            
            print("🔍 Recherche de mots-clés:\n")
            
            for label, pattern in keywords.items():
                matches = re.findall(pattern, full_text, re.IGNORECASE)
                if matches:
                    print(f"   ✅ \"{label}\": {len(matches)} occurrence(s)")
            
            # Sections importantes
            print("\n📋 Sections pertinentes:\n")
            
            # Chercher autour de "accès" ou "activation"
            access_sections = re.findall(r'.{0,200}(?:accès|activation|souscription).{0,200}', full_text, re.IGNORECASE)
            if access_sections:
                print("--- Sections sur l'accès (5 premières) ---")
                for i, section in enumerate(access_sections[:5], 1):
                    clean_section = ' '.join(section.split())
                    print(f"{i}. {clean_section}\n")
            
            # Chercher les délais
            delay_sections = re.findall(r'.{0,150}(?:délai|jour|heure|temps).{0,150}', full_text, re.IGNORECASE)
            if delay_sections:
                print("\n--- Délais mentionnés (5 premiers) ---")
                for i, section in enumerate(delay_sections[:5], 1):
                    clean_section = ' '.join(section.split())
                    print(f"{i}. {clean_section}\n")
            
            # Sauvegarder le texte complet
            output_file = file_path.replace('.pdf', '.txt')
            with open(output_file, 'w', encoding='utf-8') as f:
                f.write(full_text)
            print(f"\n💾 Texte complet sauvegardé dans: {output_file}")
            
            return full_text
            
    except Exception as e:
        print(f"❌ Erreur: {e}")
        return None

if __name__ == "__main__":
    files = [
        r"c:\Users\moros\Downloads\CGU_Legifrance_API_VF_15-12-2022 (1).pdf",
        r"c:\Users\moros\Downloads\CGU_Legifrance_API_VF_15-12-2022 (2).pdf"
    ]
    
    for file_path in files:
        import os
        if os.path.exists(file_path):
            analyze_pdf(file_path)
        else:
            print(f"⚠️  Fichier non trouvé: {file_path}")

File: scripts/analyze-cgu.js
const fs = require('fs');
const pdfParse = require('pdf-parse');

async function analyzePDF(filePath) {
  try {
    console.log(`\n📄 Analyse de: ${filePath}\n`);
    
    const dataBuffer = fs.readFileSync(filePath);
    const data = await pdfParse(dataBuffer);
    
    const text = data.text;
    
    console.log(`📊 Total pages: ${data.numpages}`);
    console.log(`📝 Caractères extraits: ${text.length}\n`);
    
    // Rechercher des mots-clés importants
    const keywords = [
      'souscription',
      'activation',
      'conditions',
      'accès',
      'OAuth',
      'authentification',
      'API',
      'délai',
      'validation',
      'acceptation',
      'CGU',
      'droit',
      'quota',
      'limite'
    ];
    
    console.log('🔍 Recherche de mots-clés:\n');
    
    keywords.forEach(keyword => {
      const regex = new RegExp(keyword, 'gi');
      const matches = text.match(regex);
      if (matches && matches.length > 0) {
        console.log(`   ✅ "${keyword}": ${matches.length} occurrence(s)`);
      }
    });
    
    // Extraire sections pertinentes
    console.log('\n📋 Sections pertinentes:\n');
    
    // Chercher les sections sur l'accès
    const accesSections = text.match(/.{0,200}(accès|activation|souscription).{0,200}/gi);
    if (accesSections) {
      console.log('--- Sections sur l\'accès ---');
      accesSections.slice(0, 5).forEach((section, i) => {
        console.log(`${i + 1}. ${section.trim()}\n`);
      });
    }
    
    // Chercher les délais
    const delaiSections = text.match(/.{0,150}(délai|jour|heure|temps).{0,150}/gi);
    if (delaiSections) {
      console.log('\n--- Délais mentionnés ---');
      delaiSections.slice(0, 5).forEach((section, i) => {
        console.log(`${i + 1}. ${section.trim()}\n`);
      });
    }
    
    // Sauvegarder le texte complet
    const outputFile = filePath.replace('.pdf', '.txt');
    fs.writeFileSync(outputFile, text);
    console.log(`\n💾 Texte complet sauvegardé dans: ${outputFile}`);
    
    return text;
    
  } catch (error) {
    console.error(`❌ Erreur: ${error.message}`);
    return null;
  }
}

async function main() {
  const files = [
    'c:\\Users\\moros\\Downloads\\CGU_Legifrance_API_VF_15-12-2022 (1).pdf',
    'c:\\Users\\moros\\Downloads\\CGU_Legifrance_API_VF_15-12-2022 (2).pdf'
  ];
  
  for (const file of files) {
    if (fs.existsSync(file)) {
      await analyzePDF(file);
    } else {
      console.log(`⚠️  Fichier non trouvé: ${file}`);
    }
  }
}

main();
